### PR TITLE
Create contributing.md

### PR DESCRIPTION
**Pull request by srijan-sivakumar**
A contributing.md is required so that we all understand the steps to be followed
for a PR and issue creation.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+## Pre-requisites.
+1. Github account ( Naturally :p )
+2. A little understanding of glusterfs is appreciated as it would come handy while dealing with the TCs and library ops.
+
+## Steps to contribute.
+
+One time effort -
+1. Fork the repository.
+2. Clone it in your local system.
+3. Add an upstream remote to fetch latest changes. `git remote add upstream git@github.com:srijan-sivakumar/redant`.git`
```
Comment (nik-redhat) [inline CONTRIBUTING.md:10]: Extra '`' 😅

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+4. Fetch the latest changes. `git fetch upstream`.
+5. Rebase it to the local clone. `git rebase upstream/main`
+
+Now, step 1-3 once done need not be repeated more than once as once they're done, you can just run the fetch and rebase just to update your local with
```
Comment (nik-redhat) [inline CONTRIBUTING.md:14]: Edit: execute the fetch command

File: CONTRIBUTING.md (hunk 1 above)
Comment (srijan-sivakumar) [reply CONTRIBUTING.md:10]: Cool, will change it :p

**Review by aujjwal-redhat — APPROVED** [review]
LGTM

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+Now, step 1-3 once done need not be repeated more than once as once they're done, you can just execute the fetch and rebase command to update your local with
+remote upstream. [ Don't forget to update your forked upstream with these latest changes in the `main` ]
+
+The following steps are to be done every time a new PR is being acted upon.
+1. Open an issue ( Bug or feature request ) detailing what is expected. Say the issue number is `XYZ`.
+2. Run steps 4-5 mentioned in the One time effort.
+3. Checkout to a new branch for making changes. `git checkout -b issueXYZ origin/main`
+4. Do the changes.
+5. Add the changed files before commit. `git add ...files..`
```
Comment (nik-redhat) [inline CONTRIBUTING.md:22]: Edit: <path-to-file(s)>

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+6. Sign your commits with `git commit -s`. This will open an editor. Add the PR description ( A one liner which will be the header for the PR )
```
Comment (nik-redhat) [inline CONTRIBUTING.md:23]: Edit: header/subject

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+followed by what was changed and why. Also add the line `Updates: #XYZ` or `Fixes: #XYZ` depending on whether this PR will fix an issue raised or
+just update a long running issue.
+7 Push the changes by running. `git push origin HEAD`
```
Comment (nik-redhat) [inline CONTRIBUTING.md:26]: Edit 7.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+Now, once the PR is opened, tag the reviewers and if the PR requires new changes follow the below steps,
+1. Fetch and rebase the local main.
+2. Checkout to the issue branch and rebase it to main. `git rebase origin/main`
+3. Do the requested changes based on review comments.
+4. This time create a new commit by signing the additions. `git commit -s`.
+5. Like the previous steps as followed for the first commit in a PR, add a basic description as to what is changed in this commit and add
+`Updates: #XYZ`
```
Comment (nik-redhat) [inline CONTRIBUTING.md:34]: I think here it can be just `git commit -s -m'<commit-msg>'`

**Review by neesingh-rh — COMMENTED** [review]
looks good to me!

**Review by neesingh-rh — APPROVED** [review]

**Review by nik-redhat — APPROVED** [review]

**Review by nishith-vihar — APPROVED** [review]

**Pull request merged**